pkg/vm/compiler/codegen.go

```diff
@@ -823,8 +858,8 @@ func CodeGen(info *buildInfo) ([]byte, error) {
 		}
 	}
 
-	if c.prog.Err != nil {
-		return nil, c.prog.Err
+	if c.Err != nil {
```

Comment:
Especially with this change. You no longer check for `prog.Err` and that's wrong.

Reply:
Why? I write to c.Err every time we have c.prog.Err. So it's one place for errors

Reply:
True, but you also have to check both then when you enter some functions.